### DISCUSSION TITLE
How do I use input slots

### DISCUSSION THREAD

**Discussion by justice1993** [jgile/vue-csv-import#79] [Q&A] [answered]
when I use the " vue-csv-input" slot,I got this error,"Cannot read properties of null (reading 'files')"

**Comment by James-Burgess** [comment, marked as answer]
Can you post an example of your `<vue-csv-import>`? I suspect you are not passing the fields prop.
Have a look at [the demo](https://github.com/jgile/vue-csv-import/blob/b20ca7daf304722f49a7fd95a157eae8dee126d4/src/App.vue#L28) and make sure you are set up correctly!

**Reply by justice1993** [reply]
I  get this error when I call change function when I want to customize the file upload button。
#### Uncaught TypeError: Cannot read properties of null (reading 'files')
```
<vue-csv-import>
...
<vue-csv-input
                name='file'
                v-slots={{
                  default: ({ file, change }) => {
                    return <input type='file' onChange={change} />
                  },
                }}
              />
</vue-csv-import>
```

**Reply by James-Burgess** [reply]
Make sure that you are passing an object to the fields prop as this is  a required field.
See [props](https://github.com/jgile/vue-csv-import/tree/b20ca7daf304722f49a7fd95a157eae8dee126d4#props)

```javascript
<vue-csv-import
        v-model="csv"
        :fields="{name: {required: false, label: 'Name'}, age: {required: true, label: 'Age'}}">
    ...
</vue-csv-import>
```
